src/Interpreters/AddDefaultDatabaseVisitor.h

```diff
@@ -121,7 +121,10 @@ class AddDefaultDatabaseVisitor
     {
         if (select.recursive_with)
             for (const auto & child : select.with()->children)
-                with_aliases.insert(child->as<ASTWithElement>()->name);
+            {
+                if (typeid_cast<ASTWithElement *>(child.get()))
```

Comment:
We cannot assume it has to be `ASTWithElement`, like `42 as ttt` is a string literal